hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfigValidator.java

```diff
@@ -50,10 +50,11 @@ public void init() {
 
   public ReplicationConfig validate(ReplicationConfig replicationConfig) {
     if (validationRegexp != null) {
-      if (!validationRegexp.matcher(replicationConfig.toString()).matches()) {
+      if (!validationRegexp.matcher(
+          replicationConfig.configFormat()).matches()) {
         throw new IllegalArgumentException("Invalid replication config " +
-            replicationConfig + ". Replication config should match the "
-            + validationPattern + " pattern.");
+            replicationConfig.configFormat() + ". Replication config " +
+            "should match the " + validationPattern + " pattern.");
```

Comment:
I am worried on displaying this validationPatten in log. Are we allowing just EC-3-2 ? without chunk size in string? ( If not, we will get another bug about this log saying " this log is confusing" :-) )
Probably we can allow in ECReplicationConfig and just take 1024K as default for chunkSize?

Reply:
We can allow 1024K but it would heavily restrict the chunk size to only a few values. Is that what we want?